.circleci/config.yml

```
@@ -312,3 +316,11 @@ workflows:
           name: "jdk17-package-site"
           image-name: *cs_img
           command: "./.ci/validation.sh package-site"
+
+  rewrite:
+    jobs:
+      - validate-with-maven-script:
+          name: "rewrite"
+          image-name: *cs_img
+          command: "./.ci/validation.sh rewrite"
+          no_output_timeout: 3h
```

hmm 3h? 
please share a reason.

---

max timeout. its running "only" 40 mins. need to build up the dom tree, then the recipes does not change the runtime a lot.

2 or 20 compisite recipes take the same 30 min warmup time. 

Locally its only 1 min vs 30 on CI, so this is quite strange.

Its only max timeout so we could start small with 1H. 

When applying all the recipes it could take 1H at the end but not much longer.

```suggestion
          no_output_timeout: 1.5h
```

---

is there way to split recipes in groups and be executed separately ?
we had same problem with pitest, that required 12hours to run all, so we split it and now we reusing parallel execution a lot.
1h job in CI is very expensive, we had is in past, always endup badly, especially when we got high traffic of PRs.

---

assuming its slow but steady it should work out. Lets start with 1H and see how it goes.

Its still roughly 10 mins buffer, as it "only" takes about 40 min on CI.

---

> is there way to split recipes in groups and be executed separately ?

yes, could move plugin call in dedicated profiles, with dedicated config/ruleset.